### ISSUE TITLE
No way to create values in Tags?

### ISSUE THREAD

**Issue by jasoncarr0** [SingingBush/SDL#6]
The SDLValue class is not public, but you require it in your addValue method. The original repo does not have this issue. There does not appear to be a way to work with these SDL trees by hand (e.g. to write SDL out), so this is a major issue.

**Comment by SingingBush** [comment]
I was focusing on my use case of parsing SDL. I'm making some changes now for version 2.0.2 and will publish shortly. Example code:

```java
Tag t = new Tag("myObject");
t.addValue(SDL.value('g'));
t.addValue(SDL.value(LocalDateTime.now()));
t.addValue(SDL.value("A \n multiline string", true));
```

**Comment by SingingBush** [comment]
I've pushed the changes to the develop branch. Would you mind cloning the repo and doing a `mvn install` from the develop branch to get _2.0.2-SNAPSHOT_ in your local mvn and try it out. I've not fully tested yet so don't want to publish to mvn central until it's been used a bit.

**Comment by jasoncarr0** [comment]
Thanks, I'll try it out. My use case involves bidirectional transformations, so I wanted to be able to create tags.

**Comment by SingingBush** [comment]
now that I've taken a little time play about with it I'm thinking about changing the API for creating SDL being more like:

```java
Tag tag = SDL.tag("thing")
    .withNamespace("my")
    .withValue(SDL.value('g'))
    .withChild(SDL.tag("child")
        .withValue(SDL.value(541L))
        .build()
    )
    .build();
```

This design feels a lot nicer to work with for me. Any thoughts? Will this be better for you?

So far these changes are just in my local branch. I'll push them when I've got a few tests in place.

**Comment by jasoncarr0** [comment]
That would be great. I think in my use case I may end up putting a facade in front anyway, but that is definitely a lot nicer to work with than the imperative method.

**Comment by jasoncarr0** [comment]
Actually one issue here is that it's a bit error-prone with strings. I think there should be two methods for SDL.value instead, one for parsing, one for going from java types, rather than an optional flag for two different behaviours on strings.

**Comment by SingingBush** [comment]
I've published 2.1.0 to maven central with the changes I proposed the other day. As well as #7 which may come in handy. I was considering having something like a _SdlWriter_ or _SdlSerializer_ class with options for outputting SDL with indentation/newlines (pretty) or minified, enabling/disabling comments, and specifying line comment style. 

```java
enum Comment {
    CPP,
    SHELL,
    LUA,
    NONE
}

SdlSerializer sdl = new SdlSerializerBuilder() // default to pretty with comments enabled as CPP style
    .pretty(true)
    .comments(LUA)
    .build();

sdl.serialize(tag);
```

however for now there's enough to get going.
